src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1436 @@
+
+#include "arkode_impl.h"
+#include "arkode_interp_impl.h"
+#include "arkode_lsrkstep_impl.h"
+
+/*===============================================================
+  Exported functions
+  ===============================================================*/
+
+void* LSRKStepCreate(ARKRhsFn fe, ARKRhsFn fi, sunrealtype t0, N_Vector y0,
+                     SUNContext sunctx)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  sunbooleantype nvectorOK;
+  int retval;
+
+  /* Check that fe is supplied */
+  if (fe == NULL)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NULL_F);
+    return (NULL);
+  }
+
+  /* Check that fi is NULL until IMEX module is ready */
+  if (fi != NULL)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "\n\nNO IMEX-LSRK support yet, set fi = NULL\n");
+    return (NULL);
+  }
+
+  /* Check for legal input parameters */
+  if (y0 == NULL)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NULL_Y0);
+    return (NULL);
+  }
+
+  if (!sunctx)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NULL_SUNCTX);
+    return (NULL);
+  }
+
+  /* Test if all required vector operations are implemented */
+  nvectorOK = lsrkStep_CheckNVector(y0);
+  if (!nvectorOK)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_BAD_NVECTOR);
+    return (NULL);
+  }
+
+  /* Create ark_mem structure and set default values */
+  ark_mem = arkCreate(sunctx);
+  if (ark_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (NULL);
+  }
+
+  /* Allocate ARKodeLSRKStepMem structure, and initialize to zero */
+  step_mem = NULL;
+  step_mem = (ARKodeLSRKStepMem)malloc(sizeof(struct ARKodeLSRKStepMemRec));
+  if (step_mem == NULL)
+  {
+    arkProcessError(ark_mem, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_ARKMEM_FAIL);
+    ARKodeFree((void**)&ark_mem);
+    return (NULL);
+  }
+  memset(step_mem, 0, sizeof(struct ARKodeLSRKStepMemRec));
+
+  /* Attach step_mem structure and function pointers to ark_mem */
+  ark_mem->step_init              = lsrkStep_Init;
+  ark_mem->step_fullrhs           = lsrkStep_FullRHS;
+  ark_mem->step                   = lsrkStep_TakeStepRKC;
+  ark_mem->step_printallstats     = lsrkStep_PrintAllStats;
+  ark_mem->step_writeparameters   = lsrkStep_WriteParameters;
+  ark_mem->step_resize            = lsrkStep_Resize;
+  ark_mem->step_free              = lsrkStep_Free;
+  ark_mem->step_printmem          = lsrkStep_PrintMem;
+  ark_mem->step_setdefaults       = lsrkStep_SetDefaults;
+  ark_mem->step_getestlocalerrors = lsrkStep_GetEstLocalErrors;
+  ark_mem->step_mem               = (void*)step_mem;
+  ark_mem->step_supports_adaptive = SUNTRUE;
```

Comment:
I like that you added method-specific constructor information to `LSRKStepSetMethod`, but as it currently stands if that routine is not called then the solver will not work correctly.  Note that in this location you specify the RKC method as the default, but you do not set the corresponding values for `LSRKmethod`, `nfusedopvecs`, `q` and `p`.  You should add those lines here so that they are set correctly in the case that a user fails to call `LSRKStepSetMethod`.